internals/plan/plan.go

```diff
@@ -974,13 +977,25 @@ func (layer *Layer) Validate() error {
 
 // Validate checks that the combined layers form a valid plan. See also
 // Layer.Validate, which checks that the individual layers are valid.
-func (p *Plan) Validate() error {
+func (p *Plan) Validate(workloads map[string]workload.Workload) error {
 	for name, service := range p.Services {
 		if service.Command == "" {
 			return &FormatError{
 				Message: fmt.Sprintf(`plan must define "command" for service %q`, name),
 			}
 		}
+		if service.Workload != "" {
+			if workloads == nil {
+				return &FormatError{
+					Message: fmt.Sprintf(`service %q cannot run in workload %q because workloads are not supported in %v`, name, service.Workload, cmd.DisplayName),
+				}
+			}
+			if _, ok := workloads[service.Workload]; !ok {
+				return &FormatError{
+					Message: fmt.Sprintf(`service %q cannot run in non-existing workload %q`, name, service.Workload),
+				}
+			}
+		}
```

Comment:
Should we enforce the rule here that if workloads are defined, uid, user, gid and group properties of the service must be unset ?